detox/src/devices/common/drivers/android/tools/HashHelper.test.js

```diff
@@ -0,0 +1,74 @@
+const HashHelper = require('./HashHelper');
+let uut;
+let adb;
+let hashxfer;
+
+describe('HashHelper', () => {
+  const mockDeviceId = '123';
+  const mockHash = 'abcdef';
+  const mockBundleId = 'com.android.test';
+
+  beforeEach(() => {
+    const ADBMock = jest.genMockFromModule('../exec/ADB');
+    const HashXferMock = jest.genMockFromModule('./HashFileXfer');
+    hashxfer = new HashXferMock();
+    adb = new ADBMock();
+
+    adb.createFileWithContent.mockImplementation( async () => jest.fn());
+
+    uut = new HashHelper(adb, hashxfer);
+  });
+
+  it('should pass arguments to adb', async () => {
+    const HASH_PATH = '/data/local/tmp/detox';
+    await uut.saveHashToRemote(mockDeviceId, mockBundleId, mockHash);
+    expect(adb.createFileWithContent).toHaveBeenCalledTimes(1);
+    expect(adb.createFileWithContent).toHaveBeenLastCalledWith(mockDeviceId, HASH_PATH, `${mockBundleId}.hash`, mockHash);
+  });
+
+  it('should return true when remoteHash and localHash match', async () => {
+    const HashXferMock = jest.genMockFromModule('./HashFileXfer');
+    hashxfer = new HashXferMock();
+    hashxfer.readHashFile.mockImplementation(() => mockHash);
+    uut = new HashHelper(adb, hashxfer);
```

Comment:
Is there a good reason to do this? `uut` already gets initialized in `beforeEach`.